components/datetime/src/datetime.rs

```diff
@@ -144,12 +164,12 @@ impl<'d> DateTimeFormat<'d> {
     pub(super) fn new<T: Into<Locale>>(
         locale: T,
         pattern: Pattern,
-        data: DataPayload<'d, 'd, DatesV1Marker>,
+        symbols_data: DataPayload<'d, 'd, DateSymbolsV1Marker>,
     ) -> Self {
         let locale = locale.into();
 
         // TODO(#257): Avoid the need for this clone by splitting the keys.
-        let symbols = DataPayload::from_owned(data.get().symbols.clone());
+        let symbols = DataPayload::from_owned(symbols_data.get().clone());
```

Comment:
Suggestion: You can delete this TODO now and just pass the `symbols_data` argument straight through. :smiley:

Reply:
Actually, ping on this suggestion.